aspnetcore/security/authentication/mfa.md

```diff
@@ -0,0 +1,560 @@
+    Layout = "/Pages/Shared/_Layout.cshtml";
+}
+```
+
+Also assign the layout for all the manage pages from the Identity pages:
+
+```cshtml
+@{
+    Layout = "_Layout.cshtml";
+}
+```
+
+### Validate the MFA requirement in the administration page
+
+The administration Razor Page validates that the user has logged in using MFA. In the `OnGet` method, the identity is used to access the user claims. The `TwoFactorEnabled` claim is checked for the value `true`. If the identity is missing this claim or is false, the page redirects to the Enable MFA page. This is possible because the user has logged in already, but without MFA.
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace IdentityStandaloneMfa
+{
+    public class AdminModel : PageModel
+    {
+        public IActionResult OnGet()
+        {
+            var claimTwoFactorEnabled = User.Claims.FirstOrDefault(t => t.Type == "TwoFactorEnabled");
+
+            if (claimTwoFactorEnabled != null && "true".Equals(claimTwoFactorEnabled.Value))
+            {
+                // You logged in with MFA, do the admin stuff
+            }
+            else
+            {
+                return Redirect("/Identity/Account/Manage/TwoFactorAuthentication");
+            }
+
+            return Page();
+        }
+    }
+}
+```
+
+### UI logic to toggle user login information
+
+An authorization policy was added at startup. The policy requires the `TwoFactorEnabled` claim with the value `true`.
+
+```csharp
+services.AddAuthorization(options =>
+{
+	options.AddPolicy("TwoFactorEnabled",
+		x => x.RequireClaim("TwoFactorEnabled", "true")
+	);
+});
+```
+
+This policy can then be used in the `_Layout` view to show or hide the **Admin** menu with the warning:
+
+```cshtml
+@using Microsoft.AspNetCore.Authorization
+@using Microsoft.AspNetCore.Identity
+@inject SignInManager<IdentityUser> SignInManager
+@inject UserManager<IdentityUser> UserManager
+@inject IAuthorizationService AuthorizationService
+```
+
+If the identity has logged in using MFA, the **Admin** menu is displayed without the warning. If the user has logged in without MFA, an icon is displayed along with the tooltip that informs the user (explaining the warning).
+
+```cshtml
+@if (SignInManager.IsSignedIn(User))
+{
+	@if ((AuthorizationService.AuthorizeAsync(User, "TwoFactorEnabled")).Result.Succeeded)
+	{
+		<li class="nav-item">
+			<a class="nav-link text-dark" asp-area="" asp-page="/Admin">Admin</a>
+		</li>
+	}
+	else
+	{
+		<li class="nav-item">
+			<a class="nav-link text-dark" asp-area="" asp-page="/Admin" 
+			   id="tooltip-demo"  
+			   data-toggle="tooltip" 
+			   data-placement="bottom" 
+			   title="MFA is NOT enabled. This is required for the Admin Page. If you have activated MFA, then logout, login again.">
+				<i class="fa fa-exclamation-triangle" aria-hidden="true"></i>
```

Comment:
Per Barry's recommendation, remove references to Font Awesome (including these CSS class names). It might make more sense to use text instead of an icon.

Reply:
@scottaddie Removed, thanks